bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -44,10 +49,19 @@
  * </ul>
  *
  * @author Yordan Mihaylov - Initial Contribution
+ * @author Victor Toni - added search capability via {@link Predicate}
  */
 public interface RuleRegistry extends Registry<Rule, String> {
 
     /**
+     * This method is used to get collection of {@link Rule}s which have the same {@link Predicate}.
+     *
+     * @param predicate to check the rules against
+     * @return collection of {@link Rule}s having matching predicate.
+     */
+    public Collection<Rule> get(Predicate<Rule> predicate);
```

Comment:
> Get method returns a single element. The name of the method should be getAll().
I will change the method name to reflect this convention.

Should the methods such as `getByTag()` and `getByTags()` be changed accordingly to `getAllByTag()` and `getAllByTags()` to support this nomenclature?